File: dotnet/EmitLog/EmitLog.cs
using System;
using RabbitMQ.Client;
using System.Text;

class EmitLog
{
    public static void Main(string[] args)
    {
        var factory = new ConnectionFactory() { HostName = "localhost" };
        using(var connection = factory.CreateConnection())
        using(var channel = connection.CreateModel())
        {
            channel.ExchangeDeclare(exchange: "logs", type: ExchangeType.Fanout);

            var message = GetMessage(args);
            var body = Encoding.UTF8.GetBytes(message);
            channel.BasicPublish(exchange: "logs", routingKey: "", basicProperties: null, body: body);
            Console.WriteLine(" [x] Sent {0}", message);
        }

        Console.WriteLine(" Press [enter] to exit.");
        Console.ReadLine();
    }

    private static string GetMessage(string[] args)
    {
        return ((args.Length > 0) ? string.Join(" ", args) : "info: Hello World!");
    }
}

File: kotlin-java-client/src/main/kotlin/EmitLog.kt
import com.rabbitmq.client.BuiltinExchangeType
import com.rabbitmq.client.ConnectionFactory


class EmitLog {
    companion object {
        const val EXCHANGE_NAME = "logs"

        fun getMessage(strings: Array<String>): String {
            return if (strings.isEmpty()) "info: Hello World!" else joinStrings(strings, " ")
        }

        private fun joinStrings(strings: Array<String>, delimiter: String): String {
            val length = strings.size
            if (length == 0) return ""
            val words = StringBuilder(strings[0])
            for (i in 1 until length) {
                words.append(delimiter).append(strings[i])
            }
            return words.toString()
        }
    }
}

fun main(args: Array<String>) {
    val factory = ConnectionFactory()
    factory.host = "localhost"
    val connection = factory.newConnection()
    val channel = connection.createChannel()
    channel.exchangeDeclare(EmitLog.EXCHANGE_NAME, BuiltinExchangeType.FANOUT)

    val message = EmitLog.getMessage(args)
    channel.basicPublish(EmitLog.EXCHANGE_NAME, "", null, message.toByteArray())
    System.out.println(" [x] Sent '$message'")

    channel.close()
    connection.close()
}

